src/Simplify.elm

```diff
@@ -5127,6 +5137,204 @@ subAndCmdBatchChecks moduleName { lookupTable, parentRange, fnRange, firstArg }
 
 
 
+-- HTML.ATTRIBUTES
+
+
+htmlAttributesClassListFalseElementError : { message : String, details : List String }
+htmlAttributesClassListFalseElementError =
+    { message = "In a Html.Attributes.classList, a tuple paired with False can be removed"
+    , details = [ "You can remove the tuple list element where the second part is False." ]
+    }
+
+
+htmlAttributesClassListChecks : CheckInfo -> List (Error {})
+htmlAttributesClassListChecks checkInfo =
+    let
+        listArg : Node Expression
+        listArg =
+            checkInfo.firstArg
+
+        getTupleWithSecond : Bool -> Node Expression -> Maybe { first : Node Expression, range : Range }
+        getTupleWithSecond bool expression =
+            case Node.value expression of
+                Expression.TupledExpression (first :: second :: []) ->
+                    if getBool bool checkInfo.lookupTable second then
+                        Just { range = Node.range expression, first = first }
+
+                    else
+                        Nothing
+
+                _ ->
+                    Nothing
+
+        singleElementListChecks : { a | element : Node Expression } -> List (Error {})
+        singleElementListChecks single =
+            case getTupleWithSecond True single.element of
+                Just tuple ->
+                    singleTrueChecks tuple
+
+                Nothing ->
+                    case getTupleWithSecond False single.element of
+                        Just _ ->
+                            [ Rule.errorWithFix htmlAttributesClassListFalseElementError
+                                checkInfo.fnRange
+                                [ Fix.replaceRangeBy (Node.range listArg) listCollection.emptyAsString ]
+                            ]
+
+                        Nothing ->
+                            []
+
+        singleTrueChecks : { a | first : Node Expression } -> List (Error {})
+        singleTrueChecks { first } =
+            [ Rule.errorWithFix
+                { message = "Html.Attributes.classList with a single tuple paired with True can be replaced with Html.Attributes.class"
+                , details = [ "You can replace this call by Html.Attributes.class with the String from the single tuple list element." ]
+                }
+                checkInfo.fnRange
+                (keepOnlyFix { parentRange = Node.range listArg, keep = Node.range first }
+                    ++ [ Fix.replaceRangeBy checkInfo.fnRange "Html.Attributes.class"
```

Comment:
Yes would be fantastic if there was some `ImportLookup = Dict ( ModuleName, String ) ModuleName` in the `CheckInfo` and a helper `toQualifiedString : ImportLookup -> ModuleName -> String -> String`.

I actually wrote this exact thing some time ago:
- https://github.com/lue-bird/generate-elm/blob/master/src/Elm/Generator.elm#L803
- https://github.com/lue-bird/generate-elm/blob/master/src/Review/Generate/Internal.elm#L178-L188
- https://github.com/lue-bird/generate-elm/blob/master/src/Review/Generate/Internal.elm#L1552-L1583

Reply:
Yeah, I agree that this information could be quite useful for other rules. I probably rewrote this logic in some other rules as well.

I'm open to adding this functionality to the core package, but this could probably use some design, with some trying it out in several rules and seeing whether the information we care about is made available

For instance, in your example for `toQualifiedString`, the return type doesn't tell you whether you need to add an import of the module is not imported (useful for https://package.elm-lang.org/packages/dillonkearns/elm-review-html-to-elm/latest/ for instance). I think there was an issue somewhere as well to know where to inject a module import, which could be useful (though that's much easier to compute now that we have the full AST easily accessible now)

Reply:
I'm still pondering whether we should probably fix this replacement module name issue before we merge this (or postpone this specific fix), because this will not work quite often. I know that this will break for my work codebase, whereas `Set` and `Dict` often will be imported without aliases.